### ISSUE TITLE
Pattern Order Affects File Counts

### ISSUE THREAD

**Issue by mmranger** [michael-doubez/filestat_exporter#21]
This is one of those edge case things that might not affect many instances.  By the way this exporter is a lifesaver for a very specific problem I have been dealing with.

Problem: The physical order of the patterns in the config file seems to affect the count.

Example: yaml
```code
...
  files:
  - patterns:
    -  "/main_dir/*"
    -  "/main_dir/Troubled_file*"
...
```

Result:
main_dir count will be accurate, but the specific file pattern within main_dir are grossly under counted. While `ls -l /main_dir/Troubled_file* | wc -l` would show around 500 of the specific files exporter would show ~0-1 sometimes 10.

Fix: yaml
```code
...
  files:
  - patterns:
    -  "/main_dir/Troubled_file*"
    -  "/main_dir/*"
...
```

Result fix:
main_dir file count and specific file pattern within main_dir seem to be more accurate.

Probably not worth a code change, but maybe a note in the documentation.

**Comment by michael-doubez** [comment]
Hello,

Indeed, that is the expacted behavior as two competing patterns will match files in different orders and thus, file matches are counted for one or the other.

The initial idea was to detect when a pattern is unused or redundant. But the variations caused by parallelism, which is an implementation detail, more or less makes it impractical. I guess, I'll make them report the number of files matching the pattern enven if it was already matched by another, it will be more intuitive.

Thanks for the feedbaack and pointing this out.

**Comment by michael-doubez** [comment]
Fixed in v0.3.6
